processor/elasticinframetricsprocessor/processor.go

```diff
@@ -57,16 +58,38 @@ func newProcessor(set processor.Settings, cfg *Config) *ElasticinframetricsProce
 
 // processMetrics processes the given metrics and applies remappers if configured.
 func (p *ElasticinframetricsProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+	newmetic := pmetric.NewMetrics()
+	rmnew := newmetic.ResourceMetrics().AppendEmpty()
+	rmscope := rmnew.ScopeMetrics().AppendEmpty()
```

Comment:
What do you think if set as Remap output the new constructed `pmetrics`? In each iteration we can append a new scoped metrics to `newmetic` with the only remapped metrics but with the same source attributes. After the metrics iteration, we can decide whether to merge both `pmetric` or just return the newly created one `override`.

I haven't test it, but something like:
```suggestion
	newmetic := pmetric.NewMetrics()

	for i := 0; i < md.ResourceMetrics().Len(); i++ {
		resourceMetric := md.ResourceMetrics().At(i)
		rm := resourceMetric.Resource()

		rmnew := newmetic.ResourceMetrics().AppendEmpty()
		rm.Attributes().CopyTo(rmnew.Resource().Attributes())

		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
			scopeMetric := resourceMetric.ScopeMetrics().At(j)
			rmscope := rmnew.ScopeMetrics().AppendEmpty()
			scopeMetric.Scope().Attributes().CopyTo(rmscope.Scope().Attributes())
			for _, r := range p.remappers {
				r.Remap(scopeMetric, rmscope.Metrics(), rm)
				if !p.cfg.Override {
					scopeMetric.CopyTo(rmscope)
				}
			}
		}
	}
	return newmetic, nil
}

```